### DISCUSSION TITLE
Flux: Not install clusterissuer, what i have to add exactly to which files ?

### DISCUSSION THREAD

**Discussion by MohamedBenighil** [fluxcd/flux2#3069] [Q&A] [answered]
Hello,

I am trying to deploy cert-manager using flux, and my repo is this [1]

And when deploye it using :

```
flux bootstrap github \
    --context=dev \
    --owner=${GITHUB_USER} \
    --repository=${GITHUB_REPO} \
    --branch=master \
    --personal \
    --path=clusters/dev
```

I got the following error:

`2022-09-02T08:30:29.259Z error Kustomization/infrastructure.flux-system - Reconciliation failed after 161.320986ms, next try in 10m0s ClusterIssuer/cert-manager/letsencrypt dry-run failed, error: no matches for kind "ClusterIssuer" in version "[cert-manager.io/v1](http://cert-manager.io/v1)"`

But when i :
  1. comment this line [2] 
  2. then deploy (push my changes), 
  3. then uncomment the same line.
  4. then deploy (push my changes), then it works fine!
 

i would like to know if you have a solution that works directly ( ie : without the 4 steps above ) ? What i have to add exactly, and to which files(s) ? 

Any idea would be really appreiated !

[1] https://github.com/MohamedBenighil/wp2
[2]
https://github.com/MohamedBenighil/wp2/blob/d4ed7b58d5bdfa671785165c31c060209c4f6457/infrastructure/certmanager/kustomization.yaml#L7

**Comment by stefanprodan** [comment, marked as answer]
Please see the docs on how to use `dependsOn`: https://fluxcd.io/flux/components/kustomize/kustomization/#kustomization-dependencies

**Reply by MohamedBenighil** [reply]
i know how to use `depedsOn ` when there is 2 Kustomization objtects, but i do not know how to use it when i have HelmRelease object and ClusterIssuer object ? any help please ?

**Reply by stefanprodan** [reply]
Place the cert-manager HelmRelease in a directory outside of the `clusters` dir, then add a Flux Kustomization that applies it.

**Comment by tuxillo** [comment]
I had a similar problem but my layout was the following:

```
├── apps
├── flux-system
├── infrastructure
│   ├── configs
│   └── controllers
└── ops
```

The problem is that when you have `flux-system` at the same level as the other directories, when Flux recurses down, it picked first the `infrastructure/configs` for whatever reason and at that point there where no ClusterIssuers installed. What I did was to move `flux-system` to a new directory `flux-root` and placed a kustomization there to pick both infrastructure directories with configs depending on controllers (like in the flux example). 

I wonder if Flux should pick any YAML files first before starting to recurse in the other directories. Or I'm just missing something.
